src/main/scala/net/codingwell/scalaguice/InjectorExtensions.scala

```diff
@@ -26,8 +26,9 @@ object InjectorExtensions {
     def instance[T: Manifest](ann: Annotation) = i.getInstance(typeLiteral[T].annotatedWith(ann))
     def instance[T: Manifest, Ann <: Annotation : Manifest] = i.getInstance(typeLiteral[T].annotatedWith[Ann])
 
-    def existingBinding[T: Manifest] = Option(i.getExistingBinding(typeLiteral[T].toKey))
-    def existingBinding[T: Manifest](ann: Annotation) = Option(i.getExistingBinding(typeLiteral[T].annotatedWith(ann)))
-    def existingBinding[T: Manifest, Ann <: Annotation : Manifest] = Option(i.getExistingBinding(typeLiteral[T].annotatedWith[Ann]))
+    def existingBinding[T: Manifest]: Option[Binding[T]] = existingBinding(typeLiteral[T].toKey)
+    def existingBinding[T: Manifest](ann: Annotation): Option[Binding[T]] = existingBinding(typeLiteral[T].annotatedWith(ann))
+    def existingBinding[T: Manifest, Ann <: Annotation : Manifest]: Option[Binding[T]] = existingBinding(typeLiteral[T].annotatedWith[Ann])
+    def existingBinding[T: Manifest](key: => Key[T]): Option[Binding[T]] = Option(i.getExistingBinding(key))
```

Comment:
It must have seemed sane at the time, looking at it again I concur with by-value.  

Ill update it.